MetaLamp/lending-pool/src/Plutus/Contracts/Endpoints.hs

```diff
@@ -83,13 +83,18 @@ start params = do
            Currency.forgeContract pkh [(Core.aaveProtocolName, 1)]
     let aave = Core.aave aaveToken
         payment = assetClassValue (Core.aaveProtocolInst aave) 1
-    ledgerTx <- TxUtils.submitTxPair $
-        TxUtils.mustPayToScript (Core.aaveInstance aave) pkh Core.LendingPoolDatum payment
+    let aaveTokenTx = TxUtils.mustPayToScript (Core.aaveInstance aave) pkh (Core.LendingPoolDatum pkh) payment
+    -- TODO how to ensure that newly minted owner token is paid to the script before someone else spends it?
+    ledgerTx <- TxUtils.submitTxPair aaveTokenTx
     void $ awaitTxConfirmed $ txId ledgerTx
 
-    let reserveMap = AssocMap.fromList $ fmap (\params -> (cpAsset params, createReserve params)) params
-    State.putReserves aave reserveMap
-    State.putUserConfigs aave AssocMap.empty
+    let reserveMap = AssocMap.fromList $ fmap (\params -> (cpAsset params, createReserve aave params)) params
+    reservesTx <- State.putReserves aave Core.StartRedeemer reserveMap
+    ledgerTx <- TxUtils.submitTxPair reservesTx
```

Comment:
Any reason the submits are separate? Although I suppose it doesn't really matter in this case.

Reply:
It does matter because during state forging we must spend and then pay the owner token back. Because it's an NFT we could not pay it back two times, so the combined transaction fails.